src/Polly.SharedSpecs/Registry/DefaultPolicyRegistrySpecs.cs

```diff
@@ -0,0 +1,238 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+using FluentAssertions;
+using Polly.Registry;
+using Polly.NoOp;
+
+namespace Polly.SharedSpecs.Registry
+{
+    public class DefaultPolicyRegistrySpecs
+    {
+        IPolicyRegistry<string, Policy> _registry;
+
+        public DefaultPolicyRegistrySpecs()
+        {
+            _registry = new DefaultPolicyRegistry();
+        }
+
+        [Fact]
+        public void Should_Allow_Adding_Policy_Using_Add()
+        {
+            Policy policy = Policy.NoOp();
+            string key = Guid.NewGuid().ToString();
+            _registry.Invoking(r => r.Add(key, policy))
+                .ShouldNotThrow();
+
+            _registry.Count.Should().Be(1);
+
+            policy = Policy.NoOp();
+            key = Guid.NewGuid().ToString();
```

Comment:
Wouldn't make any difference in behavior as I was re-initializing the variables but I see your point. I actually had done the same before but removed it later. 😞 Will make the changes.